extensions/google-translate/src/translate-form.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { Action, ActionPanel, Form } from "@raycast/api";
+import translate from "@vitalets/google-translate-api";
+import debounce from "debounce";
+import { usePreferences } from "./hooks";
+import { LanguageCode, supportedLanguagesByCode, languages } from "./languages";
+
+const TranslateForm = () => {
+  const preferences = usePreferences();
+
+  const [text, setText] = React.useState("");
+  const [fromLang, setFromLang] = React.useState<LanguageCode | string>(preferences.lang1);
+  const fromLangObj = supportedLanguagesByCode[fromLang as LanguageCode];
+  const [toLang, setToLang] = React.useState<LanguageCode | string>(preferences.lang2);
+  const toLangObj = supportedLanguagesByCode[toLang as LanguageCode];
+
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [translated, setTranslated] = React.useState<translate.ITranslateResponse["text"]>();
+
+  const doTranslate = React.useMemo(() => {
+    const debouncedTranslate = debounce(
+      async (text: string, fromLang: LanguageCode | string, toLang: LanguageCode | string) => {
+        const result = await translate(text, {
```

Comment:
It'd be great if we checked here the `text` length before wasting this request. There is a limit of 5000 chars in a single translation. So:

- Either trim `text` to max 5000 - disallow typing more OR
- Check length here and display a helpful message OR
- Create chunks (See this issue for reference: https://github.com/vitalets/google-translate-api/issues/20)

To keep it simple, I'd suggest to simply check if `text.length > 5e3` then show a toast with some message kinda "Max length (5000 chars) for a single translation exceeded."

At least the experience would be better than an HTTP error message. This can happen when pasting text from somewhere else.

Reply:
That's a good point, thanks for sharing @eddyw 💪  Could you look into this @gebeto

Reply:
Yeah, I will do it today, thank you @eddyw!